packages/modules/backup_clouds/onedrive/backup_cloud.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+import logging
+import msal
+import atexit
+import os
+import json
+from msdrive import OneDrive
+from modules.backup_clouds.onedrive.config import OneDriveBackupCloud, OneDriveBackupCloudConfiguration
+from modules.common.abstract_device import DeviceDescriptor
+from modules.common.configurable_backup_cloud import ConfigurableBackupCloud
+import pathlib
+import base64
+
+
+log = logging.getLogger(__name__)
+
+
+def get_tokens(config: OneDriveBackupCloudConfiguration) -> dict:
+    result = None
+    cache = msal.SerializableTokenCache()
```

Comment:
Es gibt die Notwendigkeit aus diesem Modul heraus den Tokencache persistent zu speichern. Hat jemand einen Tipp wie ich aus diesem Modul heraus die Konfiguration aktualisieren kann? Idee war den Tokencache dort abzulegen.
Der Cache enthält u.a. access- und refreshtoken.

Reply:
Das Thema hatte ich auch im Mercedes SoC Modul. Es gibt eine Feature Branch dafür, da es noch ein Thema mit der auth.py gibt. Da diese vom Apache aufgerufen wird, ist es nicht ideal, dass die Python Umgebung im openwb User liegt. Da kommt der Apache nicht hin, bzw. Darf da nicht ran. 
Aber die Speicherung der Token ist dort zumindest auch realisiert. 
Aktueller Workaround ist, paho.mqtt aus dem User zu deinstallieren und als root dann wieder zu installieren.

VG
Det

Reply:
Okay danke, das schaue ich mir Mal an. Ich habe alle notwendige nfunktionen um die Tokens komplett in Python zu erzeugen, meine library erstellt auch einen nutzbaren tokencache, das ist alles fertig. 
Wobei der tokencache in der config nicht ideal ist. Schöner wäre es wenn der so liegt dass der User da nicht 'dran kommt.

Ich scheitere ganz banal daran das in die config zu schreiben, habe keine heisse Idee wie ich da wieder 'ran komme.

Reply:
Ich hatte mal angefangen, ein entsprechendes Backup Modul zu machen, allerdings auf Basis von verfügbaren Linux Libs. Davfs2, s3fs, samba usw. Idee war, das entsprechende lokale oder Cloud Filesystem in das Raspbian zu mounten. Danach wäre das Backup eine einfache Copy Operation, als ob es ein lokales File wäre. Hab gerade aber zu wenig Zeit um das weiter zu treiben und nachdem ich dann gesehen habe, dass Lena auch an nem Cloud Backup arbeitet, hab ich dann nicht mehr weiter gemacht. Webdav wäre für Onedrive, Nextcloud und andere Cloud Storages nutzbar. Das Mounten würde über die vorhandenen API Keys laufen. Token Handling ist intern in den Libs gekoppelt.
S3fs setze ich beruflich auch ein und das läuft ganz gut. Davfs2 hab ich auch getestet.

VG
Det